### PR TITLE
Ikasan 2364 backfill h2 backup changes discovered while testing

### PR DESCRIPTION
**Pull request by mick-stewart73**

### CODE REVIEW

File: ikasaneip/platform/ikasan-h2-backup/readme.md [hunk 1]
```diff
@@ -13,21 +13,25 @@ not corrupt and discards the corrupted backup leaving the last good backup on th
...
+|---------------------------------------------|----------------------------------------------------------------------------------------|----------------|
+| persistence.dir                             | The base persistence directory where the database resides                              |                |
+| datasource.url                              | The H2 database connection URL                                                         |                |
+| datasource.username                         | The H2 database username                                                               |                |
+| datasource.password                         | The H2 database password                                                               |                |
+| default.ikasan.h2.backup.num.to.retain      | The number of backups to retain                                                        | 2              |
+| default.ikasan.h2.backup.cron.expression    | The Quartz cron schedule on which the backups are taken                                | 37 0/5 * * * ? |
```
Comment (andrzej-majewski) [inline ikasaneip/platform/ikasan-h2-backup/readme.md]: Seems the  default ikasan backup is at 17 sec not 37

File: ikasaneip/platform/ikasan-h2-backup/readme.md [hunk 2]
```diff
@@ -13,21 +13,25 @@ not corrupt and discards the corrupted backup leaving the last good backup on th
...
+| default.ikasan.h2.backup.on.module.shutdown | Flag to configure if a database backup should be taken when the module shuts down      | true           |
+| eai.datasource.url                          | The EAI H2 database connection URL (only if an EAI DB is necessary)                    |                |
+| eai.datasource.username                     | The EAI H2 database username (only if an EAI DB is necessary)                          |                |
+| eai.datasource.password                     | The EAI H2 database password (only if an EAI DB is necessary)                          |                |
+| eai.h2.backup.num.to.retain                 | The number of EAI database backups to retain                                           | 2              |
+| eai.h2.backup.cron.expression               | The Quartz cron schedule on which the backups are taken for the EAI DB                 | 17 0/5 * * * ? |
+| eai.h2.backup.on.module.shutdown            | Flag to configure if an EAI database backup should be taken when the module shuts down | true           |
```
Comment (andrzej-majewski) [inline ikasaneip/platform/ikasan-h2-backup/readme.md:31]: Seems the  default ikasan backup is at 37 sec not 17

File: ikasaneip/platform/ikasan-h2-backup/src/main/java/org/ikasan/backup/h2/IkasanBackupAutoConfiguration.java [hunk 3]
```diff
@@ -33,12 +40,18 @@ public class IkasanBackupAutoConfiguration {
     @Value("${default.ikasan.h2.backup.num.to.retain:2}")
     private int defaultIkasanDatabaseBackupNumToRetain;
 
-    @Value("${default.ikasan.h2.backup.cron.expression:17 0/1 * * * ?}")
+    @Value("${default.ikasan.h2.backup.cron.expression:17 0/5 * * * ?}")
```
Comment (andrzej-majewski) [inline ikasaneip/platform/ikasan-h2-backup/src/main/java/org/ikasan/backup/h2/IkasanBackupAutoConfiguration.java:43]: Double check documentation

File: ikasaneip/platform/ikasan-h2-backup/src/main/java/org/ikasan/backup/h2/IkasanBackupAutoConfiguration.java [hunk 4]
```diff
@@ -51,15 +64,29 @@ public class IkasanBackupAutoConfiguration {
...
+    @Value("${eai.h2.backup.cron.expression:37 0/5 * * * ?}")
     private String eaiIkasanDatabaseBackupCronExpression;
 
     @Value("${eai.h2.backup.on.module.shutdown:true}")
     private boolean eaiIkasanDatabaseBackupOnModuleShutdown;
 
+    @Value("${h2.db.port}")
+    private int eaiH2PortNumber;
+
+    @Value("${eai.h2.port.number.step:1000}")
```
Comment (andrzej-majewski) [inline ikasaneip/platform/ikasan-h2-backup/src/main/java/org/ikasan/backup/h2/IkasanBackupAutoConfiguration.java:76]: do we think 1000 step is good, or should be different for Ikasan

**Review by andrzej-majewski — DISMISSED** [review]

File: ikasaneip/platform/ikasan-h2-backup/src/main/java/org/ikasan/backup/h2/IkasanBackupAutoConfiguration.java (hunk 4 above)
Comment (mick-stewart73) [reply ikasaneip/platform/ikasan-h2-backup/src/main/java/org/ikasan/backup/h2/IkasanBackupAutoConfiguration.java:76]: It is configurable so in my opinion yes.

**Review by andrzej-majewski — APPROVED** [review]

**Pull request merged**